### ISSUE TITLE
Trades are not opened in parallel in backtesting mode, despite max_open_trades value >1

### ISSUE THREAD

**Issue by flxschr** [freqtrade/freqtrade#3095]
## Step 1: Have you search for this issue before posting it?

If you have discovered a bug in the bot, please [search our issue tracker](https://github.com/freqtrade/freqtrade/issues?q=is%3Aissue). 
If it hasn't been reported, please create a new issue.

## Step 2: Describe your environment

  * Operating system: Bash on Ubuntu on Windows10
  * Python Version: Python 3.6.9 :: Anaconda, Inc.
  * CCXT version: 1.23.81
  * Branch: Develop
  * Last Commit ID: 5d1b1573b76f10cadfd0f8d5436ae8cace0711b8
  
## Step 3: Describe the problem:

The bot doesn't open new trades, if the one before isn't sold/closed. No matter if I change the max_open_trades in the config.json to 10000 or to 2, it always does the same amount of trades.
I testwise changed the populate_buy_trend to an always TRUE function: dataframe.loc[(dataframe['ema5'] > 0),'buy'] = 1 , so every tick is a sell indicator, but I still see the same behaviour.

### Steps to reproduce:

  1. Backtest the Strategy003 from the sample strategy, I just changed the buy_trend (see below):
freqtrade backtesting --ticker-interval 5m -s Strategy003

### Observed Results:

*Max_open_trades = 2:*
```
Result for strategy Strategy003
=========================================================== BACKTESTING REPORT ===========================================================
|    Pair |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit BTC |   Tot Profit % |    Avg Duration |   Wins |   Draws |   Losses |
|---------+--------+----------------+----------------+------------------+----------------+-----------------+--------+---------+----------|
| ETH/BTC |     64 |          -0.05 |          -3.39 |      -0.00013592 |          -1.70 | 1 day, 12:01:00 |     56 |       0 |        8 |
| LTC/BTC |     63 |          -0.15 |          -9.19 |      -0.00036810 |          -4.60 | 1 day, 12:37:00 |     54 |       0 |        9 |
|   TOTAL |    127 |          -0.10 |         -12.59 |      -0.00050402 |          -6.29 | 1 day, 12:19:00 |    110 |       0 |       17 |
```

*Max_open_trades = 10000:*
```
Result for strategy Strategy003
=========================================================== BACKTESTING REPORT ===========================================================
|    Pair |   Buys |   Avg Profit % |   Cum Profit % |   Tot Profit BTC |   Tot Profit % |    Avg Duration |   Wins |   Draws |   Losses |
|---------+--------+----------------+----------------+------------------+----------------+-----------------+--------+---------+----------|
| ETH/BTC |     64 |          -0.05 |          -3.39 |      -0.00013592 |          -0.00 | 1 day, 12:01:00 |     56 |       0 |        8 |
| LTC/BTC |     63 |          -0.15 |          -9.19 |      -0.00036810 |          -0.00 | 1 day, 12:37:00 |     54 |       0 |        9 |
|   TOTAL |    127 |          -0.10 |         -12.59 |      -0.00050402 |          -0.00 | 1 day, 12:19:00 |    110 |       0 |       17 |
```

### Relevant code exceptions or logs:
*Here is the changed buy_trend from the strategy:*
```
    def populate_buy_trend(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
        """
        Based on TA indicators, populates the buy signal for the given dataframe
        :param dataframe: DataFrame
        :return: DataFrame with buy column
        """

        dataframe.loc[(dataframe['ema5'] > 0),'buy'] = 1


        return dataframe
```

*Here is my config.json:*
```
{
    "max_open_trades": 10000,
    "stake_currency": "BTC",
    "stake_amount": 0.004,
    "fiat_display_currency": "USD",
    "dry_run": false,
    "dry_run_wallet": 1000,
    "db_url": "sqlite:///tradesv3.sqlite",
    "unfilledtimeout": {
        "buy": 100,
        "sell": 300
    },
    "bid_strategy": {
        "ask_last_balance": 0.0
    },
    "exchange": {
        "name": "binance",
        "key": "",
        "secret": "",
        "pair_whitelist": [
            "ETH/BTC",
            "LTC/BTC",
        ],
        "pair_blacklist": [
            "DOGE/BTC"
        ]
    },
    "ask_strategy": {
        //"use_sell_signal": true,
        //"sell_profit_only": true
    },
    "telegram": {
        "enabled": false,
        "token": "",
        "chat_id": ""
    },
    "initial_state": "running",
    "internals": {
        "process_throttle_secs": 5
    }
}
```

**Comment by xmatthias** [comment]
well since you got only 2 pairs in your whitelist (`ETH/BTC` and `LTC/BTC` ) - you can only have a maximum of 2 trades in parallel (one per pair) - so `max_open_trades` has little effect unless you'd set it to 1 - which would only allow one of the 2 pairs to have an open trade0.

Opening more than one trade per pair is called position stacking - which is not supported by freqtrade at this point.

While some people would want this supported (it's been requested here #1519, and in other issues) , i'm not certain about having it as i think it can create more problems - but that's a topic for another day.

Hope this helps

**Comment by hroff-1902** [comment]
@flxschr pls verify the changes made into the documentation
